cibuildwheel/macos.py

```diff
@@ -460,15 +461,19 @@ def build(options: Options, tmp_path: Path) -> None:
                             # rosetta2 will provide the emulation with just the arch prefix.
                             arch_prefix = ["arch", "-x86_64"]
                         else:
-                            raise RuntimeError(
-                                "don't know how to emulate {testing_arch} on {machine_arch}"
-                            )
+                            msg = f"don't know how to emulate {testing_arch} on {machine_arch}"
+                            raise RuntimeError(msg)
 
                     # define a custom 'call' function that adds the arch prefix each time
-                    def call_with_arch(*args: PathOrStr, **kwargs: Any) -> None:
-                        call(*arch_prefix, *args, **kwargs)
-
-                    def shell_with_arch(command: str, **kwargs: Any) -> None:
+                    call_with_arch = functools.partial(call, *arch_prefix)
+
+                    def shell_with_arch(
+                        command: str,
+                        arch_prefix: Tuple[str, ...] = tuple(
+                            arch_prefix
+                        ),  # required to capture the value of arch_prefix now, rather than later (less confusion)
+                        **kwargs: Any,
+                    ) -> None:
```

Comment:
I have no idea what's going on here....! Is this refactor necessary?

Reply:
Same issue with capture. If you put `=thing` in your defaults, it is both explicitly captures instead of implicitly captured (good!), and it also uses the _current_ value of variable, rather than the value of the variable at the call site, which is _very_ bug prone.

Reply:
But I forgot to format this correctly in Black. 🤦

Reply:
This is now gone with the refactor.